wandb/sdk/wandb_init.py

```diff
@@ -106,11 +106,19 @@ def _handle_launch_config(settings: Settings) -> dict[str, Any]:
 
 
 class _WandbInit:
-    _init_telemetry_obj: telemetry.TelemetryRecord
-
-    def __init__(self, wl: wandb_setup._WandbSetup) -> None:
+    def __init__(
+        self,
+        wl: wandb_setup._WandbSetup,
+        telemetry: telemetry.TelemetryRecord,
+    ) -> None:
         self._wl = wl
 
+        self._telemetry = telemetry
+        """Telemetry gathered before creating a run.
+
+        After the run is created, `telemetry.context()` is used instead.
+        """
```

Comment:
i usually see inline comment in the middle of a module

Reply:
This is a convention commonly supported in IDEs to attach documentation to fields. This docstring shows up on hover in VSCode; a comment wouldn't.

Reply:
i see.. thanks!